src/commands/analyze/frameworks.ts

```diff
@@ -35,15 +39,95 @@ export async function isTypescript(contents: Record<string, string>): Promise<bo
     return false;
 }
 
-export async function getEntryfile(contents: Record<string, string>): Promise<string> {
+export async function findEntryFile(
+    componentPath: string,
+    contents: Record<string, string>,
+    patterns: RegExp[],
+    defaultFile: string,
+): Promise<string> {
+    let entryFile = await getEntryFileFromPackageJson(componentPath, contents, patterns);
+
+    if (!entryFile) {
+        entryFile = await findFileByPatterns(componentPath, patterns, EXTENSIONS);
+    }
+
+    // If no entry file is found, use the default
+    return entryFile || defaultFile;
+}
+
+async function findFileByPatterns(
+    directory: string,
+    patterns: RegExp[],
+    extensions: string[],
+): Promise<string | undefined> {
+    const entries = await fs.readdir(directory, { withFileTypes: true });
+
+    const contents: Record<string, string> = {};
+
+    // Populate contents with package.json and any files needed for direct access
+    for (const entry of entries) {
+        const fullPath = path.join(directory, entry.name);
+        if (entry.isFile()) {
+            const content = await fs.readFile(fullPath, "utf-8");
+            contents[entry.name] = content;
+        }
+    }
+
+    for (const entry of entries) {
+        const fullPath = path.join(directory, entry.name);
+
+        if (entry.isDirectory()) {
+            // Skip excluded directories
+            if (EXCLUDED_DIRECTORIES.includes(entry.name)) continue;
+
+            // Recursively search within subdirectories
+            const result = await findFileByPatterns(fullPath, patterns, extensions);
+            if (result) return result;
+        } else if (entry.isFile() && extensions.some((ext) => entry.name.endsWith(ext))) {
+            // Check if the file content matches all given patterns
+            const content = await fs.readFile(fullPath, "utf-8");
+            const allPatternsMatch = patterns.every((pattern) => pattern.test(content));
+            if (allPatternsMatch) {
+                return fullPath;
+            }
+        }
+    }
+
+    return undefined;
+}
+
+async function getEntryFileFromPackageJson(
+    directory: string,
+    contents: Record<string, string>,
+    patterns: RegExp[],
+): Promise<string | undefined> {
     if (!contents["package.json"]) {
-        return "index.mjs";
+        return undefined;
     }
 
-    // TODO Improve this - the entry file might not be defined in the package.json
-    // and it's not necessarily `index.mjs`, might be index.cjs, app.mjs, etc.
     const packageJsonContent = JSON.parse(contents["package.json"]) as PackageJSON;
-    return packageJsonContent.main || "index.mjs";
+    const mainPath = packageJsonContent.main;
```

Comment:
not sure if we want to support this, but there are multiple unofficial entrypoints in package.json: https://esbuild.github.io/api/#main-fields

Reply:
I thought about it and I'm not gonna over-engineer this.
Looking for the main entry in `package.json` it's just an heuristic to decide faster what is the main entry file.

We have a fallback to search for the main file recursively.

Thanks for suggesting it, tho' - TIL.